.github/workflows/auto-update-Dockerfiles.yml

```diff
@@ -0,0 +1,195 @@
+        shell: pwsh
+        env:
+          DOCKERFILE_PATH: ${{ env.NET_9_ARM64_Dockerfile }}
+          NEXT_VERSION: ${{ steps.get-versions.outputs.NET_9_NEXT_VERSION }}
+        run: |
+          if (-not [string]::IsNullOrEmpty("${{ env.NEXT_VERSION }}")) {
+            .\LambdaRuntimeDockerfiles\update-dockerfile.ps1 -DockerfilePath "${{ env.DOCKERFILE_PATH }}" -NextVersion "${{ env.NEXT_VERSION }}"
+          } else {
+            Write-Host "Skipping .NET 9 ARM64 update - No version detected"
+          }
+
+      # Update .NET 10 AMD64 Dockerfile
+      - name: Update .NET 10 AMD64
+        id: update-net10-amd64
+        shell: pwsh
+        env:
+          DOCKERFILE_PATH: ${{ env.NET_10_AMD64_Dockerfile }}
+          NEXT_VERSION: ${{ steps.get-versions.outputs.NET_10_NEXT_VERSION }}
+        run: |
+          if (-not [string]::IsNullOrEmpty("${{ env.NEXT_VERSION }}")) {
+            .\LambdaRuntimeDockerfiles\update-dockerfile.ps1 -DockerfilePath "${{ env.DOCKERFILE_PATH }}" -NextVersion "${{ env.NEXT_VERSION }}"
+          } else {
+            Write-Host "Skipping .NET 10 AMD64 update - No version detected"
+          }
+
+      # Update .NET 10 ARM64 Dockerfile
+      - name: Update .NET 10 ARM64
+        id: update-net10-arm64
+        shell: pwsh
+        env:
+          DOCKERFILE_PATH: ${{ env.NET_10_ARM64_Dockerfile }}
+          NEXT_VERSION: ${{ steps.get-versions.outputs.NET_10_NEXT_VERSION }}
+        run: |
+          if (-not [string]::IsNullOrEmpty("${{ env.NEXT_VERSION }}")) {
+            .\LambdaRuntimeDockerfiles\update-dockerfile.ps1 -DockerfilePath "${{ env.DOCKERFILE_PATH }}" -NextVersion "${{ env.NEXT_VERSION }}"
+          } else {
+            Write-Host "Skipping .NET 10 ARM64 update - No version detected"
+          }
+
+      # Commit changes and create a branch
+      - name: Commit and Push
+        id: commit-push
+        shell: pwsh
+        run: |
+          # Check if there are any changes to commit
+          if (git status --porcelain) {
+            git config --global user.email "github-aws-sdk-dotnet-automation@amazon.com"
+            git config --global user.name "aws-sdk-dotnet-automation"
+            $suffix=Get-Date -Format yyyy-MM-dd
+            $branch="chore/auto-update-Dockerfiles-${suffix}"
+            git checkout -b $branch
+            git add "**/*Dockerfile"
+            git commit -m "chore: Monthly ASP.NET Core version update in Dockerfiles"
+            git push origin $branch
+            
+            # Write the branch name to GITHUB_OUTPUT for use in the PR step
+            Add-Content -Path $env:GITHUB_OUTPUT -Value "BRANCH=$branch"
+            Add-Content -Path $env:GITHUB_OUTPUT -Value "CHANGES_MADE=true"
+            echo "Changes committed and pushed to branch $branch"
+          } else {
+            echo "No changes detected in Dockerfiles, skipping PR creation"
+          }
+
+      # Create a Pull Request
+      - name: Create Pull Request
```

Comment:
What would happen if we update this workflow to run daily and we don't approve and merge the created PR that same day? would this step keep creating PRs?

Reply:
fixed to use same branch